libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1136,14 +1136,21 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 	else if (valueTypes.size() != variables.size())
 	{
 		if (v050)
-			m_errorReporter.fatalTypeError(
-				_statement.location(),
-				"Different number of components on the left hand side (" +
-				toString(variables.size()) +
-				") than on the right hand side (" +
-				toString(valueTypes.size()) +
-				")."
-			);
+        {
+            // XXX If the RHS is "void", then we already have an error generated elsewhere,
```

Comment:
I am not yet sure how to fix this properly, but the error generated in `bool TypeChecker::visit(VariableDeclarationStatement const& _statement)` can be made stricter now, we can check for equality there.